server/webapp/WEB-INF/rails/webpack/views/pages/config_repos/config_repos_widget.tsx

```diff
@@ -59,32 +65,49 @@ function findPluginWithId(infos: Array<PluginInfo<any>>, pluginId: string) {
   return _.find(infos, {id: pluginId});
 }
 
-class StatusIcon extends MithrilViewComponent<{ name: string }> {
-  view(vnode: m.Vnode<{ name: string }, this>) {
-    return (
-      <div className={styles.statusIcon}>
-        {vnode.children}
-      </div>
-    );
-  }
-
-}
-
 class HeaderWidget extends MithrilViewComponent<HeaderWidgetAttrs> {
-  view(vnode: m.Vnode<HeaderWidgetAttrs>): m.Children | void | null {
+  private static readonly MAX_COMMIT_MSG_LENGTH: number = 84;
+  private static readonly MAX_USERNAME_AND_REVISON_LENGTH: number = 40;
 
+  view(vnode: m.Vnode<HeaderWidgetAttrs>): m.Children | void | null {
+    const materialUrl = vnode.attrs.repo.material().materialUrl();
     return [
-      (
-        <KeyValueTitle image={this.pluginIcon(vnode)} title={vnode.attrs.repo.id()}/>
-      ),
-      (
-        <KeyValuePair inline={true} data={new Map([
-                                                    ["Plugin Id", vnode.attrs.repo.pluginId()]
-                                                  ])}/>
-      )
+        <KeyValueTitle image={this.pluginIcon(vnode)}
+                       title={[
+                         <div class={styles.headerTitleText}><b>{vnode.attrs.repo.id()}</b></div>,
+                         <div class={styles.headerTitleText}>{materialUrl}</div>
+                       ]}/>,
+        <div>{this.latestCommitDetails(vnode.attrs.repo.lastParse())}</div>
     ];
   }
 
+  private latestCommitDetails(lastParsedCommit: ParseInfo | null) {
+    let parseStatus: m.Children = "This config repository was never parsed";
+
+    if (lastParsedCommit && lastParsedCommit.latestParsedModification) {
+      const comment = this.trimToLength(lastParsedCommit.latestParsedModification.comment, HeaderWidget.MAX_COMMIT_MSG_LENGTH);
+      const username = this.trimToLength(lastParsedCommit.latestParsedModification.username, HeaderWidget.MAX_USERNAME_AND_REVISON_LENGTH);
+      const revision = this.trimToLength(lastParsedCommit.latestParsedModification.revision, HeaderWidget.MAX_USERNAME_AND_REVISON_LENGTH);
+
+      parseStatus = (
+        <div class={styles.headerTitleText}>
+          {comment}
+          <div><b>{username}</b> | {revision}</div>
+        </div>
+      );
+    }
+
+    return parseStatus;
+  }
+
+  private trimToLength(str: string, length: number): string {
```

Comment:
Could we possibly use [_.truncate](https://lodash.com/docs/4.17.11#truncate) out of the box?